### ISSUE TITLE
"Unsupported filesystem type: module" message during Quasar startup

### ISSUE THREAD

**Issue by garyb** [quasar-analytics/quasar#2836]
This might just be a problem of "wrong error message", but currently I'm seeing this in the startup log.

```
Warning: Failed to mount '/spark-local/Untitled Workspace 10.slam/.tmp/outc2ee99b4-af3f-476c-ae19-b73e3fd095ec/' because 'Invalid mount config, '[module] CREATE FUNCTION cardc2ee99b4af3f476cae19b73e3fd095ec(:fieldName)
  BEGIN
    foo := lat; (select ((t).coords)[0] as lat, ((t).coords)[1] as lon, (t).location from `/spark-local/locations.json` as t order by :fieldName ASC)
  END', because: Unsupported filesystem type: module'.
```

(Yes I realise the query is gibberish :wink:)

This could be because spark-local isn't available via plugins yet, or the various #2813 issues (I'm trying on Windows). Thought I'd open an issue for it anyway in case it really is broken. I'll retry all this on a Mac next.

**Comment by djspiewak** [comment]
Does this happen with other backends, or just Spark Local?

**Comment by alissapajer** [comment]
This may be  https://github.com/quasar-analytics/quasar/issues/2370.

**Comment by garyb** [comment]
@djspiewak I don't think I have any other modules mounted right now, I'll try and rig something up to test it under a mongo mount and report back.

@alissapajer interesting, it could be - although it occurs whenever I try to start Quasar, so it's writing a corrupted metastore at least. Full startup log in case its of use:

```
Read web config from quasar\config.json
log4j:WARN No appenders could be found for logger (com.zaxxer.hikari.HikariConfig).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Using metastore: jdbc:h2:C:\Users\Gary\AppData\Local\quasar\quasar-metastore.db
Warning: Failed to mount '/mongoq/' because 'Invalid mount config, '[mongodbq] mongodb://localhost:63174', because: Unsupported filesystem type: mongodbq'.
Warning: Failed to mount '/mongo/' because 'Invalid mount config, '[mongodb] mongodb://localhost:63174', because: Unsupported filesystem type: mongodb'.
Warning: Failed to mount '/spark-local/' because 'Invalid mount config, '[spark-local] /Home/Documents/Projects/quasar-test-starter/tmp/spark-local/', because: Unsupported filesystem type: spark-local'.
Warning: Failed to mount '/spark-local/Untitled Workspace 10.slam/.tmp/outc2ee99b4-af3f-476c-ae19-b73e3fd095ec/' because 'Invalid mount config, '[module] CREATE FUNCTION cardc2ee99b4af3f476cae19b73e3fd095ec(:fieldName)
  BEGIN
    foo := lat; (select ((t).coords)[0] as lat, ((t).coords)[1] as lon, (t).location from `/spark-local/locations.json` as t order by :fieldName ASC)
  END', because: Unsupported filesystem type: module'.
Server started listening on port 80
Press Enter to stop.
```

The mongoq mount in there hasn't worked for a while obviously, so I'm used to seeing that one!

**Comment by wemrysi** [comment]
@garyb What version are you using?

Also, we should really output the version to the console when starting the web server.

**Comment by garyb** [comment]
This was v24.0.15

**Comment by garyb** [comment]
@djspiewak this also failed for a module under mongo:

```
Warning: Failed to mount '/mongo/test/Untitled Workspace 8.slam/.tmp/out2d3a054d-31ca-444e-958c-f683b52900ac/' because 'Invalid mount config, '[module] CREATE FUNCTION card2d3a054d31ca444e958cf683b52900ac(:city)
  BEGIN
    (select * from `/mongo/test/smallZips` where ((city) = (:city)))
  END', because: Unsupported filesystem type: module'.
```

Again though, this is without a mongo plugin either due to #2813. I'll see what happens on my Mac now.

**Comment by garyb** [comment]
Ok, I've run this on my Mac now, with the mongo connector and everything mounted, and it still raises the error on startup:

```
Warning: Failed to mount '/mongo/test/Untitled Workspace.slam/.tmp/out0abed8ac-2b29-43c4-a1ba-8f942e5219dc/' because 'Invalid mount config, '[module] CREATE FUNCTION card0abed8ac2b2943c4a1ba8f942e5219dc(:city)
  BEGIN
    (select * from `/mongo/test/smallZips` where ((city) = (:city)))
  END', because: Unsupported filesystem type: module'.
```

But, the module is still actually there if I go and look in the filesystem after restarting Quasar.

Initially I tried this by running and older Quasar first and then upgrading. After I saw the error occurred there I deleted my metastore, and started completely fresh with the v24.0.15 release, set it all up again, and indeed the error still occurs.

So at least it's not quite as bad as it appears, since the module mount is actually working, it's just odd that the error is printed during startup.
